### DISCUSSION TITLE
Can't save data

### DISCUSSION THREAD

**Discussion by GalinaBublik** [filamentphp/filament#6072] [Help] [answered]
I have a problem with saving data. 

This is local test project.
At first i createt resource for Page
after that created model and migration for Page.
After some manipulation pages start to saved correct (filled $fillable, many times clear chache)

Than i want to test Builder::make('content')->blocks
Did Roll back my migration, correct file of migration to 
`Schema::create('pages', function (Blueprint $table) {
            $table->id();
            $table->integer('author_id');
            $table->string('title');
            $table->text('excerpt')->nullable();
            $table->json('content')->nullable()->default(json_encode([]));
            $table->string('slug')->unique();
            $table->string('image')->nullable();
            $table->text('meta_title')->nullable();
            $table->text('meta_description')->nullable();
            $table->text('meta_keywords')->nullable();
            $table->enum('status', Page::$statuses)->default(Page::STATUS_INACTIVE);
            $table->string('password')->nullable();
            $table->text('layout')->nullable();
            $table->timestamp('deleted_at')->nullable();
            $table->timestamps();
        });`
Mifrated, tried to create new Page - but got an error.
https://screenrec.com/share/7Q4PkzOa3D (Main Page created by phpmyadmin )
Did  php artisan optimize:clear - not help

Deleted 'content' from $fillable, php artisan optimize:clear
Saved Content to DB {"type":"heading","data":{"content":"tjdkdfjb","level":"h2"}}
But not have another error
![image](https://user-images.githubusercontent.com/10061993/229050452-3f074ea6-a514-46c2-98f0-78a9eaccf064.png)
And can't create new page ((
https://screenrec.com/share/OEjsLRqdT2

Please help!
Maybe some command for renew model setting or some detailed information about Builder field settings

My migration  
`$table->json('content')->nullable()->default(json_encode([]));`
My model
`public function getContentAttribute($value)
    {
        if(is_array($value)){
            return $value;
        }
        return json_decode($value);
    }

`

**Comment by danharrin** [comment, marked as answer]
You don't need an accessor. You need an `array` cast. https://laravel.com/docs/10.x/eloquent-mutators#array-and-json-casting

**Reply by GalinaBublik** [reply]
Thank you so much! 
Commented out getContentAttribute 
Added protected $casts = [
        'content' => 'json',
    ];

Pages working correct.
